lib/BuildSystem/SwiftTools.cpp

```diff
@@ -536,6 +541,50 @@ class SwiftCompilerShellCommand : public ExternalCommand {
     
     return true;
   }
+
+  std::string previousOutput;
+  ssize_t numBytes = 0;
+
+  void partialOutputAvailable(StringRef partialOutput) {
+    previousOutput.append(partialOutput);
+
+    do {
+      if (numBytes == 0) {
+        auto newLinePos = previousOutput.find("\n");
+        // Don't even have the first line yet.
+        if (newLinePos == std::string::npos) {
+          return;
+        }
+
+        StringRef numBytesString = previousOutput.substr(0, newLinePos);
+        if (StringRef(numBytesString).getAsInteger(10, numBytes)) {
+          numBytes = 0;
+          // Couldn't get number of bytes.
+          // Probably due to a compiler crash or bad output?
```

Comment:
We should probably do something else here, even if it just a `// FIXME: Need logging infrastructure.` to get a supported API for logging this kind of "unexpected, but possible" error.